torch/_higher_order_ops/schema.py

```diff
@@ -122,11 +130,23 @@ def add_arg(
     def add_output(self, output: Any) -> None:
         self.example_outputs.append(output)
 
+    def add_schema_tree_spec(self, *args: Any, **kwargs: Any) -> None:
+        """schema tree spec is the tree spec from flattening all inputs to the hop with pytree.tree_flatten
+        Since torch.FunctionSchema only have proper mutation/alias support for flattened inputs, we need
+        to store the tree spec in order to reconstruct the inputs to the hop.
+        """
+        self.schema_tree_spec = pytree.tree_flatten((args, kwargs))[1]
```

Comment:
I feel like this feature should only be used with the HOPs that truly need it (like cond). We shouldn't try to flatten the args to invoke_subgraph for example.

Reply:
The motivation is that I don't want to unnecessarily flatten things like list[int]. flattening has performance implications

Reply:
Sounds good. I'll make it None for hops that take flattened inputs.

Reply:
And assert that a schema_tree_spec is provided when we schema_gen() if the inputs are not flattened.